apps/tasks/views.py

```diff
@@ -1,36 +1,71 @@
 
-    @action(detail=False, serializer_class=DisplayTaskSerializer)
-    def display_my(self, request, *args, **kwargs):
-        instance = Task.objects.filter(user=self.request.user)
-        serializer = DisplayTaskSerializer(instance, many=True)
-        return Response(data=serializer.data)
-
-    @action(detail=False, serializer_class=TaskSerializer)
-    def completed(self, request, *args, **kwargs):
-        complete_task = Task.objects.filter(status="Completed")
-        serializer = TaskSerializer(complete_task, many=True)
-        return Response(data=serializer.data)
-
-    @action(methods=["PATCH"], detail=False, serializer_class=TaskAssignSerializer)
-    def assign(self, request, *args, **kwargs):
-        Task.objects.filter(pk=self.request.data["id"]).update(user=self.request.data["user"])
-        return Response(status.HTTP_200_OK)
+    @action(detail=False, methods=['PUT'], serializer_class=TaskAssignSerializer)
+    def assign(self, request):
+        Task.objects.filter(pk=self.request.data['id']).update(assigned_to=self.request.data['assigned_to'])
+        task = Task.objects.get(pk=self.request.data['id'])
+        send_user_email(
+            subject="You have ben assigned a new task!",
+            message=f"You have ben assigned a new task!\n The new task is \"{task.title}\".",
+            from_email=settings.EMAIL_HOST_USER,
+            recipient_list=[request.user.email]
+        )
+        return Response({}, status=http.HTTPStatus.OK)
+
+    def perform_update(self, serializer):
+        super().perform_update(serializer)
+        if serializer.instance.status == Task.Status.COMPLETED:
+            comments = Comment.objects.filter(task=serializer.instance)
+            for comment in comments:
+                send_user_email(
+                    subject="Your task, that was commented is completed!",
+                    message=f"You have just executed a task!\n The completed task is \'{serializer.instance.title}\'.",
+                    from_email=settings.EMAIL_HOST_USER,
+                    recipient_list=[comment.user.email]
+                )
+
+
+class CommentViewSet(ModelViewSet):
+    serializer_class = TaskCommentSerializer
+    queryset = Comment.objects.all()
+    permission_classes = (IsAuthenticated,)
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(user=request.user)
+        headers = self.get_success_headers(serializer.data)
+        task = Task.objects.get(pk=serializer.data['task'])
+        send_user_email(
+            subject='Your task got a new comment!',
+            message=f'The task \"{task.title}\" got a new comment :\n {serializer.data["text"]}',
+            from_email=settings.EMAIL_HOST_USER,
+            recipient_list=[request.user.email]
+        )
+        return Response(serializer.data, status=201, headers=headers)
```

Comment:
Aici mai bine ar fi ca sa folosesti metoda `perform_create`, ca o sa trimiti email, in loc ca sa rescrii `create`